LDMP/jobs/mvc.py

```diff
@@ -192,7 +202,37 @@ def updateEditorGeometry(
         editor.setGeometry(option.rect)
 
 
-class DatasetEditorWidget(QtWidgets.QWidget, WidgetDatasetItemUi):
+class DatasetActions:
+    job: models.Job
+
+    def load_dataset(self):
+        manager.job_manager.display_job_results(self.job)
+
+    def open_job_directory(self):
+        log(f"Open directory button clicked for job {self.job.params.task_name!r}")
+        job_directory = manager.job_manager.get_job_file_path(self.job).parent
+        # NOTE: not using QDesktopServices.openUrl here, since it seems to not be
+        # working correctly (as of Jun 2021 on Ubuntu)
+        openFolder(str(job_directory))
+
+    def delete_dataset(self):
+        message_box = QtWidgets.QMessageBox()
+        message_box.setText(
+            f"You are about to delete job {self.job.params.task_name!r}")
+        message_box.setInformativeText("Confirm?")
+        message_box.setStandardButtons(
+            QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.Cancel)
+        message_box.setDefaultButton(QtWidgets.QMessageBox.Cancel)
+        message_box.setIcon(QtWidgets.QMessageBox.Information)
+        result = QtWidgets.QMessageBox.Res = message_box.exec_()
+        if result == QtWidgets.QMessageBox.Yes:
+            log("About to delete the dataset")
+            for path in self.job.results.local_paths:
+                layers.delete_layer_by_filename(str(path))
+            manager.job_manager.delete_job(self.job)
+
+
+class DatasetEditorWidget(QtWidgets.QWidget, DatasetActions,  WidgetDatasetItemUi):
```

Comment:
Please rename this to `DatasetEditorDialogue` and make it a subclass of `QtWidgets.QDialog`.

Also, as mentioned above:

- put this code in another file, not on `jobs.mvc`
- `DatasetActions` should not be a superclass to this one. Merge it into this